app/src/main/java/com/xinto/opencord/ui/screen/Chat.kt

```diff
@@ -176,8 +175,31 @@ private fun ChatScreenLoaded(
                                     onClick = {},
                                     onLongClick = { showBottomDialog = true }
                                 ),
+                            reply = message.isReply.let { if (it) it else null }?.letComposable { _ ->
+                                val referencedMessage = message.referencedMessage
+                                if (referencedMessage != null) {
+                                    WidgetMessageAvatar(
+                                        modifier = Modifier.size(20.dp),
+                                        url = referencedMessage.author.avatarUrl
+                                    )
+                                    WidgetMessageReplyAuthor(author = referencedMessage.author.username)
+                                    if (referencedMessage.contentNodes.isNotEmpty()) {
+                                        WidgetMessageReplyContent(
+                                            text = render(
+                                                nodes = referencedMessage.contentNodes,
+                                                renderContext = null
+                                            ).toAnnotatedString()
+                                        )
+                                    }
+                                } else {
+                                    Text("Unknown message")
```

Comment:
This could probably attempt to fetch the referenced message manually before this ever happens

Reply:
Probably should be done after we implement CacheManager